### ISSUE TITLE
Cannot build using parms for GPT4, 1106-preview for westus region.

### ISSUE THREAD

**Issue by evogelpohl** [Azure-Samples/azure-search-openai-demo#1239]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ x] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
I want to use GPT-4, not 3.5 for my deployment. Not vision. 

I've confirmed I have quota to deploy OpenAI GPT 4 in `westus`.

I've edited the list of regions in the main.bicep to include `westus` for `param openAiResourceGroupLocation`

I've edited the main.bicep file to change the deployment to: 

`param chatGptModelName string = (openAiHost == 'azure') ? 'gpt-4' : 'gpt-4'`
`param chatGptModelVersion string = '1106-preview'`

This is according to the table @ https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#gpt-4-and-gpt-4-turbo-preview. 

Run `azd up` & select `westus` for my openai deployment, and I get this error:

`ERROR: deployment failed: failing invoking action 'provision', error deploying infrastructure: deploying to subscription:

Deployment Error Details:
InvalidTemplateDeployment: The template deployment 'openai' is not valid according to the validation procedure. The tracking id is 'e2d3af54-fbfc-49c7-9c40-67cda6153479'. See inner errors for details.
DeploymentModelNotSupported: The model 'Format: OpenAI, Name: gpt-4, Version: 1106-preview' of account deployment is not supported.

TraceID: 8874dd9589a4b507224477a8611845ed`

azd version 1.5.1 (commit 3856d1e98281683b8d112e222c0a7c7b3e148e96)

**Comment by pamelafox** [comment]
Are you able to create that deployment from the Azure Portal as well? Sometimes there are availability issues that we haven't accounted for in our Bicep.

**Comment by evogelpohl** [comment]
> Are you able to create that deployment from the Azure Portal as well? Sometimes there are availability issues that we haven't accounted for in our Bicep.

Yes. I can manually create the OpenAI instance (in the portal, for westus). Then, within the studio, deploy the GPT-4, ver 1106-Preview. 

Then, I can define the `azd env ...` vars as documented specifying a pre-established OAI service, rg, and chat model by name. The `azd up` process fails w/ the error of ~ 'you cannot change models` or similar. 

I'm in a 'catch-22'
If I deploy to `eastus`, we have no quota or permission to use any gpt-4 model at all and the permission form doesn't allow you to specific the region, 

If I deploy to `westus`, it doesn't support deploying gpt-4 using a version that this project accepts. Thus, i'm stuck w/ 3.5-turbo.
